src/CodeViewer/index.tsx

```diff
@@ -43,21 +81,51 @@ const CodeViewer: React.FunctionComponent<ICodeViewerProps> = ({
     );
   }
 
-  const markup = parseCode(value, lang, showLineNumbers);
+  const actualClassName = cn(Classes.CODE_EDITOR, className, `language-${lang || 'unknown'}`, {
+    [`${Classes.CODE_EDITOR}--inline`]: inline,
+    [`${Classes.CODE_EDITOR}--line-numbers`]: showLineNumbers,
+  });
+
+  if (markup === null) {
+    return (
+      <pre className={actualClassName} {...rest}>
+        {value}
+      </pre>
+    );
+  }
 
   return (
-    <pre
-      className={cn(Classes.CODE_EDITOR, className, `language-${lang || 'unknown'}`, {
-        [`${Classes.CODE_EDITOR}--inline`]: inline,
-        [`${Classes.CODE_EDITOR}--line-numbers`]: showLineNumbers,
-      })}
-      {...rest}
-    >
-      {markup ? markup.map(astToReact()) : value}
-    </pre>
+    <AutoSizer>
+      {({ height, width }) => (
+        <FixedSizeList
```

Comment:
Note - the code will not be copy-able anymore, because we'll only render a small chunk of it.
Is this okay?

Reply:
Got rid of it eventually, since search would be broken either, and this is likely to be annoying.